### PR TITLE
First part of ERC734

### PR DESCRIPTION
**Pull request by filiplazovic**
I'm not sure i understood the part with keys correctly. When function e.g. `addKey` is called i'm checking if the key is hash of the caller address, which i saw from the [example](https://github.com/OriginProtocol/origin-playground/blob/master/contracts/KeyHolder.sol#L67) from `origin-playground`.

EDIT:  After a little more research i saw [here](https://github.com/tyleryasaka/identity-proposals/blob/master/contracts/KeyManager.sol#L89) that key can be added on behalf of somebody else who signed this transaction(which is not implemented here). Other than that, is there any other way this key is going to be used?

### CODE REVIEW

**Comment by tyleryasaka** [comment]
@filiplazovic the `addKeySigned` method [here](https://github.com/tyleryasaka/identity-proposals/blob/master/contracts/KeyManager.sol#L89) is *not* part of the proposed ERC734 standard. I was just experimenting with that. It does not need to be implemented here.

File: contracts/package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "truffle": "^4.1.14",
     "web3": "^1.0.0-beta.36",
     "chai": "^4.2.0",
-    "openzeppelin-solidity": "^2.0.0"
+    "openzeppelin-solidity": "^2.0.0",
+    "web3-utils": "^1.0.0-beta.36"
```
[the remark this answers is not in the document]
Comment (filiplazovic) [reply contracts/package.json]: Oh, sorry, i didn't see that we had web3 1.0 included

**Comment by filiplazovic** [comment]
@tyleryasaka  This can be reviewed

**Comment by tyleryasaka** [comment]
Hey @filiplazovic , this is looking really good!

To me, there are two approaches here:

1) Provide a single `setKey` method, and assume that conversion from bits to human-friendly integers will be handled externally (off-chain). I would expect this approach to be more efficient in terms of gas usage.

2) Provide separate `addKey` and `removeKey` methods, and convert from human-friendly integers to bitmask representations on-chain. (In this case, I don't think `setKey` is needed.) This is the approach you are taking here.

I personally lean toward the first approach. I don't think it is as important to have a clean solidity interface. I would rather optimize for gas. But if you prefer the second approach, I'm fine with that for now.

**Comment by tyleryasaka** [comment]
@filiplazovic I made a few tweaks so that the interface matches [this interface](https://github.com/ERC725Alliance/erc725/blob/master/docs/ERC-734.md#solidity-interface) as closely as possible.

**Pull request merged**